### DISCUSSION TITLE
should Joomla 4 api work while astroid installed?

### DISCUSSION THREAD

**Discussion by BoxSo** [templaza/astroid-framework#218] [Q&A] [answered]
We tried creating articles by using the joomla 4 api and the server throws an error 500 cause of an "undefined constant":

`{ "errors": { "code": 500, "title": "Internal server error", "detail": "Error: Undefined constant \"Astroid\\Helper\\ASTROID_JOOMLA_VERSION\" in /<<localPath>>/httpdocs/libraries/astroid/framework/library/astroid/Helper/Client.php:207\nStack trace:\n#0 /<<localPath>>/httpdocs/plugins/system/astroid/astroid.php(54): Astroid\\Helper\\Client->onContentPrepareForm()\n#1 /<<localPath>>/httpdocs/libraries/src/Plugin/CMSPlugin.php(279): plgSystemAstroid->onContentPrepareForm()\n#2 /<<localPath>>/httpdocs/libraries/vendor/joomla/event/src/Dispatcher.php(486): Joomla\\CMS\\Plugin\\CMSPlugin->Joomla\\CMS\\Plugin\\{closure}()\n#3 /<<localPath>>/httpdocs/libraries/src/Application/EventAware.php(107): Joomla\\Event\\Dispatcher->dispatch()\n#4 /<<localPath>>/httpdocs/libraries/src/MVC/Model/FormBehaviorTrait.php(169): Joomla\\CMS\\Application\\WebApplication->triggerEvent()\n#5 /<<localPath>>/httpdocs/administrator/components/com_content/src/Model/ArticleModel.php(1034): Joomla\\CMS\\MVC\\Model\\FormModel->preprocessForm()\n#6 /<<localPath>>/httpdocs/libraries/src/MVC/Model/FormBehaviorTrait.php(107): Joomla\\Component\\Content\\Administrator\\Model\\ArticleModel->preprocessForm()\n#7 /<<localPath>>/httpdocs/administrator/components/com_content/src/Model/ArticleModel.php(466): Joomla\\CMS\\MVC\\Model\\FormModel->loadForm()\n#8 /<<localPath>>/httpdocs/libraries/src/MVC/Controller/ApiController.php(420): Joomla\\Component\\Content\\Administrator\\Model\\ArticleModel->getForm()\n#9 /<<localPath>>/httpdocs/libraries/src/MVC/Controller/ApiController.php(316): Joomla\\CMS\\MVC\\Controller\\ApiController->save()\n#10 /<<localPath>>/httpdocs/libraries/src/MVC/Controller/BaseController.php(672): Joomla\\CMS\\MVC\\Controller\\ApiController->add()\n#11 /<<localPath>>/httpdocs/libraries/src/Dispatcher/ApiDispatcher.php(61): Joomla\\CMS\\MVC\\Controller\\BaseController->execute()\n#12 /<<localPath>>/httpdocs/libraries/src/Component/ComponentHelper.php(355): Joomla\\CMS\\Dispatcher\\ApiDispatcher->dispatch()\n#13 /<<localPath>>/httpdocs/libraries/src/Application/ApiApplication.php(405): Joomla\\CMS\\Component\\ComponentHelper::renderComponent()\n#14 /<<localPath>>/httpdocs/libraries/src/Application/ApiApplication.php(113): Joomla\\CMS\\Application\\ApiApplication->dispatch()\n#15 /<<localPath>>/httpdocs/libraries/src/Application/CMSApplication.php(294): Joomla\\CMS\\Application\\ApiApplication->doExecute()\n#16 /<<localPath>>/httpdocs/api/includes/app.php(53): Joomla\\CMS\\Application\\CMSApplication->execute()\n#17 /<<localPath>>/httpdocs/api/index.php(31): require_once('...')\n#18 {main}" } }`

Seems that ASTROID has no constant for Joomla Version defined in context of Joomla 4.2 API. This breaks the API completely.

**Comment by sonvnn** [comment, marked as answer]
Please follow this issue https://github.com/templaza/astroid-framework/issues/216
